docs/03-language-guide/14-equality.md

```diff
@@ -1,6 +1,131 @@
 ---
 title: Equality
 id: equality
+description: A reference guide to equality in Wing.
+keywords: [Winglang equality, Wing equality, equals, identity]
 ---
 
-TODO (for now, see the [language spec RFC](../999-contributors/999-rfcs/2023-06-12-language-spec.md))
+Checking for equality is performed with the `==` operator in Wing. It returns `true` if the two values are equal, and `false` otherwise.
+
+The main difference between equality in JavaScript and Wing is that `==` in Wing is not allowed to compare values of different types. For example, `1 == "1"` is not allowed in Wing, and will result in a compile-time error.
+
+The execution phase ([preflight or inflight](../02-core-concepts/01-preflight-and-inflight.md)) that a value was created in does not affect its equality. For example, a value created in preflight can be equal to a value created in inflight.
```

Comment:
This weirdly feels like it's too complicated to be correct, but I can't think of any issues, so 👍

Reply:
Yeah, it's a little optimistic, for sure :-)

One question I had was if you have two instances of a class created in preflight, and then you lift them into inflight, can you still safely check them for equality? In any case, I'm sure we'll figure out in due time

Reply:
I feel like this paragraph is a pretty advanced statement and should likely go at the bottom of this section (perhaps also the one about JavaScript).